src/transformers/models/clvp/configuration_clvp.py

```diff
@@ -0,0 +1,489 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" CLVP model configuration"""
+
+import copy
+import os
+from typing import TYPE_CHECKING, Union
+
+
+if TYPE_CHECKING:
+    pass
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+CLVP_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "susnato/clvp_dev": "https://huggingface.co/susnato/clvp_dev/resolve/main/config.json",
+}
+
+
+class CLVPTextConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`CLVPTextModel`]. It is used to instantiate a CLVP
+    text encoder according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the text encoder of the CLVP
+    [susnato/clvp_dev](https://huggingface.co/susnato/clvp_dev) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        vocab_size (`int`, *optional*, defaults to 256):
+            Vocabulary size of the CLVP text model. Defines the number of different tokens that can be represented by
+            the `inputs_ids` passed when calling [`CLVPTextModel`].
+        hidden_size (`int`, *optional*, defaults to 768):
+            Dimensionality of the encoder layers and the pooler layer.
+        intermediate_size (`int`, *optional*, defaults to 1536):
+            Dimensionality of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        projection_dim (`int`, *optional*, defaults to 768):
+            Dimensionality of the text projection vector.
+        num_hidden_layers (`int`, *optional*, defaults to 20):
+            Number of hidden layers in the Transformer encoder.
+        num_attention_heads (`int`, *optional*, defaults to 12):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        hidden_act (`str` or `function`, *optional*, defaults to `"gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
+            `"relu"`, `"selu"` and `"gelu_new"` `"quick_gelu"` are supported.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-5):
+            The epsilon used by the layer normalization layers.
+        attention_dropout (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        dropout (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the feed-forward layers in [`CLVPMLP`].
+        use_attention_bias (`bool`, *optional*, defaults to `False`):
```

Comment:
Are `use_attention_bias`, `use_glu_in_ff`, `ff_post_act_layer_norm`, `use_pre_branch_norm`, `use_post_main_norm` and `norm_type` every any different to their default values? For a `transformers` port, we shouldn't copy the original code one-for-one, but instead try to make our port as clean and as "slick" as possible. This means that all code that is unnecessary should be removed (i.e. code that is never used). This also means that if you notice that some configuration parameters are never changed for the pretrained model, then we don't add them into our config but just hardcode them.